### ISSUE TITLE
Custom Tracking

### ISSUE THREAD

**Issue by PhilColbert** [Ysurac/FlightAirMap#296]
Thank you for taking time to reply to our email - added this to issues as you asked for.
>
> We feed to various other live tracking sites, however 3d would be
> fantastic !
>
> We have this for the groundstations :-
>
> http://www.airwhere.co.uk/gspositions.php
>
> These for the pilots - its raining here today so no one is flying, the
> project is mainly UK based but expanding over the world everyday  (
> updated once a second )
>
> http://www.airwhere.co.uk/pilots.php
>
> And for the pilot details :-
>
> http://www.airwhere.co.uk/pilotdetails.php?pilot=222
>
> However all this can be changed and any format thats required can be made.
>
> All changes that would make - definitely made available, its your
> project, I would always like to contribute :)

**Comment by Ysurac** [comment]
In latest commit there is a beta support of AirWhere source. But need more testing and some config parameters needed are not yet in install script (to display pilots instead of owners).
All should be done next week. 

The same code will be used for Open Glider Network (glidernet).

**Comment by PhilColbert** [comment]
Wow, nice one ! :)

Its it worth installing and trying or wait until next week.

Thanks again, really looking forward to it :)

**Comment by Ysurac** [comment]
Wait next week, this will be easier.

**Comment by PhilColbert** [comment]
Thanks, do you have a donation paypal or something towards the project ? :)

**Comment by Ysurac** [comment]
It's on https://www.flightairmap.fr/ buttons before *An application with many features*. But wait until next week when you will run it :)

**Comment by Ysurac** [comment]
I have **403 Forbidden** *You don't have permission to access / on this server.* on http://www.airwhere.co.uk/. It's seems my IP is banned.

**Comment by PhilColbert** [comment]
Sorry, I just thought someone had left their browser window open for days
and days, didnt know it was you testing, will unban again.... its happened
a few times.

All done.

Phil

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by PhilColbert** [comment]
Its been a bad time for testing as its not been flyable for paragliders
here !

Hopefully be a better day this week :)

Phil.

**Comment by Ysurac** [comment]
All should work, even if I don't really had many paragliders to test for now.

**Comment by PhilColbert** [comment]
Wow nice one, so whats the best way forward, download the repo and install
on my server, how to setup the config for airwhere?

Hopefully we will have a flying day soon to test properly.

Regards

Phil.


<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by Ysurac** [comment]
Yes, add all dependencies, clone repo, then run *install/index.php*, this will check if all dependencies are here.
As source put airwhere website and format *airwhere*. For 3D view you need a Bing map key.
You can set some configs options like enable checkbox *No airlines check and display*.
Then submit the form, this will download 3D models if enabled, create and populate DB,...
You can always go to *install/index.php* and change some default options, this is also needed when you update FlightAirMap.

**Comment by PhilColbert** [comment]
Ok thanks, really appreciate this , so exciting :)

How do we go about adding new models , ie paragliders ? I guess there arent
any in there already?

Thanks

Phil.

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by Ysurac** [comment]
I added one.

**Comment by PhilColbert** [comment]
Ha, good man!

Just got to build a couple of flight units today for pilots and then will
make a start.

Do you fly at all ?

Phil.

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by Ysurac** [comment]
I don't fly at all for now. But I like to track flights.

**Comment by PhilColbert** [comment]
Well if your ever in the UK and you would like a tandem cross country
flight across some of the most beautiful parts of our country, please come !

You can watch your live 3d flight while your flying on a tablet ? ! :)


Phil.

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by Ysurac** [comment]
Should be possible to watch while flying but this will not be exactly real time and need a quite good internet connection to download map or satellite view :)

**Comment by PhilColbert** [comment]
I know :) I just thought it would be funny !

I was wondering do you take each tracking point and jump to that point when
it arrives or run the system in slight lag and extrapolate points and
create a smooth transition ?

Phil.

**Comment by Ysurac** [comment]
I extrapolate. It's why it's not really real time.

**Comment by PhilColbert** [comment]
Though so, will get on with it today, thanks again, amazing work!

Phil.

**Comment by PhilColbert** [comment]
Just wondered, I thought you used cesium for 3d tracking ? Not bing ?

Just thought of the cost for bing thats all.

Thanks

Phil.

**Comment by PhilColbert** [comment]
Sorry to keep bothering you but :)

not sure how the $globalSources variable should be setup ?

$globalSources = array(array('host' => 'www.airwhere.co.uk','port' =>
'80','name' => 'airwhere','format' => 'airwhere','sourcestats' =>
TRUE,'noarchive' => FALSE));

Thanks

Phil.

**Comment by Ysurac** [comment]
I use Cesium for 3D tracking but Cesium used Bing for some data. Seems to be fixed in latest version. You can try without any key.

**Comment by Ysurac** [comment]
Should be ok. But *sourcestats* can't really be used because you aggregate ground station result.

**Comment by PhilColbert** [comment]
Maps loading ok , but no aircraft being detected and the app isnt reading
pilots.php off the server.

http://www.airwhere.co.uk/live-tracking-3d/

Any ideas ?

Thanks

**Comment by Ysurac** [comment]
You need to have the script *scripts/daemon-spotter.php* running.

**Comment by PhilColbert** [comment]
Ok, thanks, didnt know, is there anything else I need to run :)

Ok, its now got 3 aircraft which is correct, does the SCAN MODE and then
checks for the 3 aircraft and then

This is repeated many times.


error : SQLSTATE[42S22]: Column not found: 1054 Unknown column
'location_id' in 'where clause'error :

Getting there.. :)

**Comment by Ysurac** [comment]
* You need to run *scripts/update_db.php* at least one time, this will download 3D models. Then you should put it in a cron job to run every day.
* *scripts/daemon-spotter.php* must be running as a daemon via init script or as cron job but **$globalDaemon** have to be set to **FALSE**.
* I forget to add the *location_id* column in sql. Sorry. You have to insert *db/source_location.sql* again or add a *location_id* integer column in *source_location* table.

**Comment by Ysurac** [comment]
*$globalMapRefresh* is too low, this update really too often. It's not needed.

**Comment by PhilColbert** [comment]
Ok, I thought this was the timing to update the map with new positions ? 30
then ?

Phil.

**Comment by Ysurac** [comment]
Yes 30 should be ok. It's **$globalMinFetch** to fetch new position. But it's also not needed to fetch every seconds. A paraglider is not moving so fast :)

**Comment by PhilColbert** [comment]
Ok , please dont say sorry, all new software has issues to fix :)

Running update_db, its taking quite a while.... sat on

updating NOTAM...NOTAM from FlightAirMap website : Download...Gunzip...Add
to DB...

Ground stations have appeared - step in the right direction, its good to
fetch every seconds, I update live tracking 2d on our server every 2 , nice
to see people thermalling :)

Phil.

**Comment by Ysurac** [comment]
Update can be slow, lot of data to add. Check that **$globalTransaction** is set to **TRUE**, else this can be really slow.

This can make the DB really busy, 10 should be enough. **$globalLiveInterval** should be high value, I 
use 800. This let paraglider on map even if no new data was seen since this time.

**Comment by PhilColbert** [comment]
Ok, would like to leave last position seen on the map for the day.... then
people can see where the last known position was, to come find the
paraglider if hes in the area :)

So

script scripts/daemon-spotter.php running.
$globalDaemon FALSE

scripts/update_db.php - cron daily

$globalMapRefresh 30
$globalMinFetch 10 ?
$globalLiveInterval 800 - or do I need to go higher to have it on for the
day?




Thanks

Phil.

**Comment by Ysurac** [comment]
All values should be ok, but it's not really possible to keep a paraglider visible on the map for a day. It's really for live mapping. I can make some tests and changes to support that.

**Comment by PhilColbert** [comment]
Sure, I understand - thank you for offering to do more functionality, it
will be the first live 3d tracking site for paragliders :)

For example

Just finishing a new live tracking 2d for us with google maps.

Paraglider just goes grey with out of signal, green for walking, blue for
flying and red - in trouble  ( request for help )

We will be adding lots of functions later ,for example , being able to send
the internet information via the ground stations, so areas with no
internet, pilots will still be able to communicate, all exciting stuff ,
just a lot of work for me :)

Phil.

**Comment by Ysurac** [comment]
I try to add more functionality and fix bugs quite often. I have a lot more to add.

You have a problem with the **.htaccess** if you are running Apache, or you don't use the *install/flightairmap-nginx-conf.include* if you use Nginx, because URL rewriting is not working.

**Comment by PhilColbert** [comment]
Ok, will look into that, where have you seen that not work then I have an
example to fix ?

Thanks

**Comment by Ysurac** [comment]
In all menu. Like http://www.airwhere.co.uk/live-tracking-3d/pilot

**Comment by PhilColbert** [comment]
Ahh ok, yes I have seen that, will check next thing :)

**Comment by PhilColbert** [comment]
For some reason the aircraft detected is just spinning waiting, it was
updating before with a number but stopped now, any ideas?

does update_db run in a loop or stop when its done , been running for an
hour now...

Sat here :-

Updating aircraft's owners...
owner from FlightAirMap website : Download...Gunzip...Add to DB...Done

Thanks

**Comment by PhilColbert** [comment]
Thats only on 2d, it finds 3 on 3d.

Altitude is being found incorrectly so the model is under the terrain i
think, when i zoom far out I can just see a glider model, how to change to
paraglider model  ?

 ------- Check Country for  with latitude : 53.699821 and longitude :
-2.262032.... FOUND :  ---------------
DATA : hex :  - ident :  - ICAO : PARAGLIDER - Departure Airport : NA -
Arrival Airport : NA - Latitude : 53.699821 - Longitude : -2.262032 -
waypoints :  - Altitude : 2 - Heading : 164 - Speed : 1 - Departure Airport
Time :  - Arrival Airport time :  - Source Name :

Altitude - 2 ?

Phil.

**Comment by Ysurac** [comment]
In 2D it's because rewriting is not working, when menu will work 2D should work too.

The altitude is correct but there is a problem with STK terrain. No idea why for now. No problem with ellipsoid.

The model you see is the default model, it's because models are not yet downloaded. Update script should work and exit when done.

**Comment by PhilColbert** [comment]
Ok, thanks, the server isnt responding to rewrite for some reason, and git
clone didnt pull down .htaccess either...

Altitude for this paraglider is 750 feet, not 2 , sorry dont understand.

Wow thats some update script ! :)

Phil.

**Comment by Ysurac** [comment]
in pilot.php altitude is in feet or something else ?

**Comment by PhilColbert** [comment]
My apologies, in metres.

so its 221 metres , do you go by 2 meaning 200 metres ?

Phil.

**Comment by Ysurac** [comment]
In debug yes, it's flight level (FL) altitude.
I fixed altitude in latest commit, I used feet instead of meters.

**Comment by PhilColbert** [comment]
Ok , so am I in debug, sorry for all the questions, all very new to me, are
you basing the altitude on FL, would be more useful for us to be in metres/
feet ?

the pilots.php is in metres... I can change one to feet if need be , whats
easiest for you , apologies.

Fixed rewrite now.

2d is working with the paragliders appearing, very dark blue ones, is that
correct - where do I change the icons if need be?


Thanks

**Comment by Ysurac** [comment]
I fixed in latest commit, now altitude will be ok.
Users can change icon in settings sidebar, else it's **$globalAircraftIconColor**

**Comment by PhilColbert** [comment]
Ok, just changed the spotter file to work, coming out ok now I think.

Are you running just in flight level then , not metres, 100metres is a lot
of height difference when flying near terrain or with other pilots ?

Phil

**Comment by Ysurac** [comment]
True, I will add real altitude to DB ASAP. It was not a problem for commercial flights, but now it is.

**Comment by Ysurac** [comment]
It's 100 feet, so 30,5m. But I will and use real altitude in DB.

**Comment by PhilColbert** [comment]
Thanks, whats your job by the way, great you have time and motivation for a
project like this, its hard keeping motivated sometimes I find!

update is trying to download aircraft models now.

I assume the paraglider model will download, theres an aircraft flying
slightly low over my house ! :) haha

Phil.

**Comment by Ysurac** [comment]
Real altitude is now in latest commit. You will need to do a **git pull** and run **install/index.php** again to update database schema.

**Comment by PhilColbert** [comment]
Nice one ! :)

Will do.

Phil.

**Comment by PhilColbert** [comment]
Last time I ran the install it didnt come up with the instructions to run
the deamon etc.

This time it did, hence me not knowing the first time

Thanks

Phil

**Comment by Ysurac** [comment]
I added some instructions, it wasn't always showing to run *daemon-spotter.php*.

**Comment by PhilColbert** [comment]
haha, that would be why then :)


Thanks

Phil

**Comment by Ysurac** [comment]
*$globalURL* is not correctly set. You should set it to **/live-tracking-3d**.

**Comment by PhilColbert** [comment]
I know, i just reinstalled and missed it out, its already sorted , just a
typo

Looking better, 2 planes above where I am, terrain seems a bit out.....
225meters here, and at 227m the aircraft are at least 300metres above ?

Am I doing something wrong ?

How to get the paraglider models in ?

Thanks

**Comment by Ysurac** [comment]
It display the previous altitude value, not the current interpolated one. I will look into that.

Do you have **paraglider.glb** in *models* directory ?

**Comment by PhilColbert** [comment]
Nope, the update script is still running, been going for 4 hours so far !

Where can I download it from please ?

Thanks

**Comment by Ysurac** [comment]
If you only download it, this will not work.
It's too slow. What is displayed ?

**Comment by PhilColbert** [comment]
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-2008.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-2007.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-2006.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-2005.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-2004.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-2003.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-2002.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-2001.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-2000.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1999.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1998.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1997.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1996.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1995.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1994.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1993.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1992.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1991.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1990.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1989.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1988.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1987.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1986.csv
Import
/home/fugazi/air-ware/public_html/live-tracking-3d/require/../install/tmp/cr-1985.csv

**Comment by Ysurac** [comment]
ok it's importing flights accidents.
I will need to try to reproduce this problem.

**Comment by PhilColbert** [comment]
No worries, will just wait, its a long install script is half a day ! :)

Thanks

**Comment by PhilColbert** [comment]
Any idea why the aircraft displayed are 300metres above the terrain ?

Their altitude is correct on the screen, but they are way way above where
they should be

Thanks

Phil.

**Comment by Ysurac** [comment]
No idea, I will make some check. I changed to use altitude as altitude above ground but I would need to translate that to height above ellipsoid...

**Comment by PhilColbert** [comment]
That would make sense, if you removed terrain height, it would be onto the
ground.

Paragliders have appeared, thanks !!

Looks great - will look aronud for some more paraglider models :)

Thanks

**Comment by PhilColbert** [comment]
Would it be possible to make sure the models stayed in a vertical position,
the angle of attack doesnt change when flying a paraglider unlike a plane ?

Thanks


<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by PhilColbert** [comment]
By the way running the deamon-spotted, sends mysql into 500% cpu usage with
these queries.

 Query   |  346 | Sending data | SELECT EXTRACT(month FROM date) AS month,
EXTRACT(year FROM date) AS year, SUM(fatalities) as count  |
Query   |  344 | Sending data | SELECT EXTRACT(month FROM date) AS month,
EXTRACT(year FROM date) AS year, SUM(fatalities) as count  |
Query   |  315 | Sending data | SELECT EXTRACT(month FROM date) AS month,
EXTRACT(year FROM date) AS year, SUM(fatalities) as count

Is there anyway i can turn this off easily, I would guess its the size of
the accident tables, seems to not finish the query before starting another
one.

Box is a quad xeon....

Thanks

**Comment by Ysurac** [comment]
This query is run when stats are updated using *scripts/update_db.php*. How often do you run it ? Check that there is no multiple scripts that are running (even if this shouldn't be possible).

**Comment by PhilColbert** [comment]
I have only run it the once at the start, not run it again.

Just come back to the machine after a days flying and found 15 of those
processing running ......

Strange, will keep an eye on it

Thanks

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#m_4727537549698528557_DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by PhilColbert** [comment]
Sorry to ask, but did you manage to fix the altitude issue with the aircraft appearing 100-200 metres above the terrain ?

Thanks

**Comment by Ysurac** [comment]
Should be better in latest commit.
On what is based your altitude ? GPS (should be over ellipsoid) ? barometric ? AMSL (above mean sea level) ? something else ?

By default I use GPS altitude over ellipsoid. Now I fix altitude by checking ground level, I will try to make a better algorithm soon.

**Comment by PhilColbert** [comment]
That aircraft icon in question, when you click on it is says 235m, which is
amsl, it looks roughly around 600m amsl.

Thats the height of the house pretty much.

Will just get the new files when I get time and try again

Thanks

Phil.

**Comment by Ysurac** [comment]
In the version you use, it's 235m above ground level. Now it will be 235m above ellipsoid if it's not under ground level. I will try to find a way to get sea level.
